extensions/amp-story-auto-ads/0.1/test/test-amp-story-auto-ads.js

```diff
@@ -396,5 +399,54 @@ describes.realWin(
         });
       });
     });
+
+    describe('development mode', () => {
+      it('should immediately insert and navigate to ad page', async () => {
+        const storeService = await Services.storyStoreServiceForOrNull(win);
+        const storeStub = sandbox.stub(storeService, 'get');
+        storeStub
+          .withArgs(StateProperty.CURRENT_PAGE_ID)
+          .returns('story-page-0');
+        storeStub.callThrough();
+
+        const storyImpl = new MockStoryImpl(storyElement);
+        storyElement.getImpl = () => Promise.resolve(storyImpl);
+        await addStoryPages(doc, storyImpl);
+
+        adElement.id = 'i-amphtml-demo-1';
+        adElement.setAttribute('development', '');
+        const config = {
+          'a4a-conversion': true,
+          src: '/examples/amp-story/ads/app-install.html',
+          type: 'fake',
+        };
+        addStoryAutoAdsConfig(adElement, config);
+        await autoAds.buildCallback();
+        await autoAds.layoutCallback();
+        addCtaValues(autoAds, 'SHOP', 'https://example.com');
+
+        const adPageElement = doc.querySelector('#i-amphtml-ad-page-1');
+        const insertSpy = sandbox.spy(storyImpl, 'insertPage');
+        const dispatchStub = sandbox.spy(storyEvents, 'dispatch');
+
+        const ampAd = doc.querySelector('amp-ad');
+        ampAd.signals().signal(CommonSignals.INI_LOAD);
+        await macroTask();
+
+        expect(insertSpy).calledWith('story-page-0', 'i-amphtml-ad-page-1');
+        const payload = {
+          'targetPageId': 'i-amphtml-ad-page-1',
+          'direction': NavigationDirection.NEXT,
```

Comment:
glad you have this! that makes sure a test failure when the constant is changed.